### ISSUE TITLE
rule "no-new" examples incorrect

### ISSUE THREAD

**Issue by boggddan** [eslint/eslint#8810]
**What did you do? Please include the actual source code causing the issue.**

The example rule  [no-new](http://eslint.org/docs/rules/no-new) specifies

> Examples of correct code for this rule:
```js
/*eslint no-new: "error"*/

Thing();
```
but, if without **new** appear error
```js
class Test {
}
Test();
```

```
Uncaught TypeError: Class constructor Test cannot be invoked without 'new'
    at 1.self-2a6e1f9328605df6da3ed9b2e72f8eefb89782162e8c81b0ac2cedcfa2ac8ebe.js?body=1:3
```

**Comment by platinumazure** [comment]
Hi @boggddan, thanks for the issue.

Class constructors must be invoked with `new`, as you have seen. Our documentation should definitely be updated to reflect this.

It would be nice if we could detect if a constructor is a class constructor rather than a function constructor, but that would be impossible when using constructors defined in other files (either as globals, or `require`d in CommonJS/Node, or `import`ed ES6 modules) since ESLint only runs on one file at a time. So I'm not sure we can enhance the rule to try to detect those reliably. I'm thinking maybe we should add a note in the documentation stating that we don't recommend using this rule if the code in question is using ES6 classes.

**Comment by boggddan** [comment]
Hi @platinumazure, thanks for the quickly reply.

**Comment by not-an-aardvark** [comment]
My understanding was that the rule is encouraging the user to avoid writing constructors with side-effects. So it might not be possible to fix the problem just by removing `new`, but it's still a valid issue nonetheless.

**Comment by boggddan** [comment]
Maybe, The solution would be to add a parameter with the value of a regular expression, so that the developer could name his classes with the addition of a prefix or suffix and thereby separate where the class and where the function

**Comment by not-an-aardvark** [comment]
> The solution would be to add a parameter with the value of a regular expression, so that the developer could name his classes with the addition of a prefix or suffix and thereby separate where the class and where the function

Personally, I think that is out of scope for the `no-new` rule. It's just intended to prevent calling constructors with side-effects, not to disallow calling constructors that would result in an error being thrown.

**Comment by not-an-aardvark** [comment]
Closing because I think the documentation is correct as-is.
